plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -242,7 +242,7 @@ internal fun SpdxDocument.addExtractedLicenseInfo(licenseFactProvider: LicenseFa
     val nonSpdxLicenses = allLicenses.filter { SpdxConstants.isPresent(it) && SpdxLicense.forId(it) == null }
 
     val extractedLicenseInfo = nonSpdxLicenses.sorted().mapNotNull { license ->
-        licenseFactProvider.getLicenseText(license)?.let { text ->
+        licenseFactProvider.getLicenseText(license)?.takeIf { it.isNotBlank() }?.let { text ->
```

Comment:
Shouldn't this rather be implemented on the provider API level, or at least for the composite provider instead? Otherwise other providers are not tried if a previous one returns a blank license.

Reply:
Yes, I thought about it, but this would be a bigger refactoring, because the `hasLicense` checks have to be re-implemented.